src/test/tools/convert_xml_to_yaml/xml2yaml.yaml

```diff
@@ -0,0 +1,40 @@
+options:
+  stoptime: 3600
+topology:
+- graphml: |
+    <?xml version="1.0" encoding="utf-8"?><graphml xmlns="http://graphml.graphdrawing.org/xmlns" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://graphml.graphdrawing.org/xmlns http://graphml.graphdrawing.org/xmlns/1.0/graphml.xsd">
+      <key attr.name="packetloss" attr.type="double" for="edge" id="d6" />
+      <key attr.name="latency" attr.type="double" for="edge" id="d5" />
+      <key attr.name="packetloss" attr.type="double" for="node" id="d4" />
+      <key attr.name="countrycode" attr.type="string" for="node" id="d3" />
+      <key attr.name="bandwidthdown" attr.type="int" for="node" id="d2" />
+      <key attr.name="bandwidthup" attr.type="int" for="node" id="d1" />
+      <key attr.name="ip" attr.type="string" for="node" id="d0" />
+      <graph edgedefault="undirected">
+        <node id="isp">
+          <data key="d0">0.0.0.0</data>
+          <data key="d1">2251</data>
+          <data key="d2">17038</data>
+          <data key="d3">US</data>
+          <data key="d4">0.0</data>
+        </node>
+        <edge source="isp" target="isp">
+          <data key="d5">50.0</data>
+          <data key="d6">0.01</data>
+        </edge>
+      </graph>
+    </graphml>
+plugin:
```

Comment:
It's a little odd that we have list values with singular keys. e.g. `plugin` instead of `plugins`, `host` instead of `hosts`. The singular form makes sense in xml, where it's referring to a single item in that list, but is a little confusing in yaml, where it's referring to the whole list.

At the risk of making the mapping between xml and yaml a little less consistent, I'd lean towards remapping to/from the the singular/plural forms.

@robgjansen wdyt?
@florian-vuillemot how difficult would this be to do?

Reply:
I agree with @sporksmith: plurals for the `yaml` lists makes more sense to me too.

Reply:
Hum 3 choices:
- Hard code each key and the plural
- Add an 's' to each key automatically
- Use both methods if there is a key which plural is different from adding a simple 's' at the end (process => processes)

What do you prefer ?

Reply:
For the sake of simplicity and consistency, I'd lean towards just using a hard-coded explicit mapping for all of these; i.e. the first option.

Reply:
Where could I find all keys to map ?

Reply:
I think you got them all:
- plugin -> plugins
- host -> hosts
- process -> processes

Also, "node" was renamed to "host", and "application" was renamed to "process", but we still technically accept "node" and "application" as valid inputs. So I suppose you should add these too:
- node -> nodes
- application -> applications

Reply:
And if you're curious, the parsing code starting at root level is here:
https://github.com/shadow/shadow/blob/master/src/main/core/support/configuration.c#L733

Reply:
I could also do:
    node -> hosts
    application -> processes
This could help to remove old input.

Reply:
Sure, OK. But then you would have to spell out an explicit reverse mapping from YAML -> XML to make sure that "hosts" in YAML maps to "host" in XML (and not "node" in XML). So I think you will have to actually type out the `YAML_TO_XML` table rather than doing this:

```
YAML_TO_XML = {v: k for k, v in XML_TO_YAML.items()}
```